(82)(68) validation and testing + Display friends

based-kwl opened the pull request:
- Validation class
- Added many tests for current API routes (67 tests total)
- Merged PR #73

mnashat commented on __test__/student.spec.js:
```diff
@@ -19,133 +20,395 @@ afterAll(async () => {
-  //works as cleanup of previous test too.
-  it("delete a Student ", async () => {
+  //Delete student tests
+  it("Delete a Student ", async () => {
```
After deleting a student (user1.email in this case), we should test that this student is also removed from the friend list of other students (user2) in this case.

based-kwl replied:
Thus functionality does not exist. It has to be implemented first to test it

mnashat replied:
Noted, thanks for letting me know.

mnashat commented on __test__/student.spec.js:
```diff
@@ -19,133 +20,395 @@ afterAll(async () => {
-    let expected = "user_" + random4
-    let arr = ["user_" + random2 + "@gmail.com", "user_" + random3 + "@gmail.com", "user_" + random0 + "@gmail.com"]
+  //Friend request tests
+  let FriendRequest, FriendRequest2;
```
Please change variable names to start with lowercase characters.

mnashat reviewed (changes requested):
Great job on the tests, well done! Just a few comments/changes for consistency. Check the comments.

Also, we should rename `Student_test_data.js` to `student_test_data.js` for consistency with other file names.

Thank you!

mnashat reviewed (approved):
Great job on the tests! Approved.

guaj commented:
your branch is adding code that I deleted (duplicates). You should probably rebase and remove code push in the following files :
- friendRequest repository
- friends.js

Also implemented e2e tests to check if your features is breaking functionnality with main. All tests are failing right now, so need to check these out before merge.

based-kwl commented on src/repository/friendRequestRepository.js:
```diff
@@ -2,14 +2,40 @@ const FriendRequest = require('../models/friendRequest.model');
+     * @param {string} requestID : id of the request to delete
+     * @returns {Promise<unknown>}
+     */
+    static async deleteFriendRequest(requestID) {
```
This is not duplicate.findOneAndDelete is not the same as deleteOne. findOneAndDelete will return the document deleted while deleteOne will only return the status, i.e. number of documents deleted (1)

based-kwl replied:
delete uses deleteOne and deleteFriendRequest uses findOneAndDelete

based-kwl commented:
> your branch is adding code that I deleted (duplicates). You should probably rebase and remove code push in the following files :
> 
> * friendRequest repository
> * friends.js
> 
> Also implemented e2e tests to check if your features is breaking functionnality with main. All tests are failing right now, so need to check these out before merge. In the readMe file in the FE repo you'll find the procedure on how to run these.
> 
> <img alt="image" width="566" src="https://user-images.githubusercontent.com/60197474/204827005-4af63753-661d-4a9c-98b4-4cb95db6b3af.png">

resolved, see FE PR for more details

based-kwl commented:
![image](https://user-images.githubusercontent.com/59633304/205208670-3bdca35f-7671-4c0d-a64f-79aef7cf6253.png)
Results of e2e tests with https://github.com/based-kwl/SemesterPlanner-client/pull/94
